influence_benchmark/retroactive_evaluator/run_retroactive_evals.py

```diff
@@ -0,0 +1,177 @@
+
+    save_name = run + "_gpt"
+    print(f"Saving results_df as {save_name}.pkl")
+    save_pickle(results_df, f"{save_name}.pkl")
+
+
+def evaluate_runs_gpt(
+    runs: List[str],
+    backend_config: Dict[str, Any],
+    env_config_path: Path,
+    max_trajs_per_env: int,
+    max_iter: Optional[int],
+    load: bool,
+    save: bool,
+    run_dir_prefix: Path,
+) -> Dict[str, pd.DataFrame]:
+    results_df_dict = {}
+    processes = []
+    mp.set_start_method("spawn")
+
+    for run in runs:
+        print(f"Starting process for run {run}.")
+        p = mp.Process(
+            target=evaluate_single_run_gpt,
+            args=(run, backend_config, env_config_path, max_trajs_per_env, max_iter, load, save, run_dir_prefix),
+        )
+        processes.append(p)
+        p.start()
+
+    for p in processes:
+        p.join()
+
+    # Since we saved the results within each process, we can load them here
+    for run in runs:
+        save_name = run + "_gpt"
+        results_df = pd.read_pickle(f"{save_name}.pkl")
+        results_df_dict[run] = results_df
+
+    return dict(results_df_dict)
+
+
+def evaluate_runs_hf(
+    runs: List[str],
+    backend_config: Dict[str, Any],
+    env_config_path: Path,
+    devices: Optional[List[int]] = None,
+    batch_size: Optional[int] = None,
+    max_trajs_per_env: int = 4,
+    max_iter: Optional[int] = None,
+    load: bool = False,
+    save: bool = False,
+    run_dir_prefix: Path = Path("/nas/ucb/micah/Influence-benchmark/data/trajectories"),
+) -> Dict[str, pd.DataFrame]:
+    results_df_dict = {}
+
+    for run in runs:
+        run_dir = run_dir_prefix / run
+        metrics = metrics_by_run(run)
+        print(f"Evaluating run {run}.")
+
+        evaluator = HFRetroactiveEvaluator(
+            run_path=run_dir,
+            backend_config=backend_config,
+            metrics=metrics,
+            batch_size=batch_size,
+            devices=devices,
+            env_config_path=env_config_path,
+            max_trajs_per_env=max_trajs_per_env,
+        )
+
+        results_df = evaluator.evaluate_run(load=load, save=save, max_iter=max_iter)
+
+        save_name = run
+        print(f"Saving results_df as {save_name}.pkl")
+        save_pickle(results_df, f"{save_name}.pkl")
+
+        results_df_dict[run] = results_df
+
+    return results_df_dict
+
+
+if __name__ == "__main__":
+    runs = [
+        "KTO_nudging_therapist_veto-09_08_123317",
+        "KTO_nudging_therapist_veto-09_12_092627",
+        # "KTO_tickets_veto_with_tools-09_09_113234",
+        "weak-therapist3t-env-09_12_221249",
+    ]
+    # Needs to be provided if "preference" is one of the metrics
+    env_config_path = None
+    gpt = True
+    load = False
+    save = False
+    max_trajs_per_env = 1
+    max_iter = 3
+    # To allow for absolute paths instead of using PROJECT_DATA
+    run_dir_prefix = Path("/nas/ucb/adhyyan/Influence-benchmark/data/trajectories")
```

Comment:
Maybe we can discontinue this because you can't actually read stuff in other people's directories, you have to first copy them over

Reply:
you can change it to use the path in root?